### PR TITLE
StopIteration changes

### LINKED ISSUE #647: use try and except block around StopIteration or avoid StopIteration completely
`StopIteration` is deprecated from python3.6+, https://docs.python.org/3.6/whatsnew/3.6.html#deprecated-python-behavior 

As per https://www.python.org/dev/peps/pep-0479/ we should not use `raise StopIteration` on generators.
See #636

### PR DESCRIPTION
**Pull request by dhivyaaxim**
This PR resolves #647 
use try and except block around StopIteration or avoid StopIteration completely

### CODE REVIEW

File: src/python/turicreate/toolkits/style_transfer/_sframe_loader.py [hunk 1]
```diff
@@ -125,9 +125,13 @@ def provide_data(self):
         return self._provide_data
 
     def _next(self):
-
+
```
Comment (shantanuchhabra) [inline src/python/turicreate/toolkits/style_transfer/_sframe_loader.py]: Let's not add leading whitespace.

File: src/python/turicreate/toolkits/style_transfer/_sframe_loader.py [hunk 2]
```diff
@@ -125,9 +125,13 @@ def provide_data(self):
         return self._provide_data
 
     def _next(self):
-
+        
         if self.cur_epoch == self.num_epochs or len(self.sframe) == 0:
-            raise StopIteration
+            import sys
+            if(sys.version_info.major == 3 and sys.version_info.minor == 7):
```
Comment (shantanuchhabra) [inline src/python/turicreate/toolkits/style_transfer/_sframe_loader.py]: What is the significance of 3 and 7? Could we add a comment explaining the magic numbers or make those into variables?

Comment (TobyRoseman) [reply src/python/turicreate/toolkits/style_transfer/_sframe_loader.py]: It seems `StopIteration` is still included in Python 3.7. It just get automatically converted to a RuntimeError.
From https://docs.python.org/3.7/library/exceptions.html -
```
Changed in version 3.7: Enable PEP 479 for all code by default: a StopIteration error raised in a generator is transformed into a RuntimeError.
```
So I don't think this change is necessary. @dhivyaaxim - are we missing something?

Comment (dhivyaaxim) [reply src/python/turicreate/toolkits/style_transfer/_sframe_loader.py]: @TobyRoseman  Thanks for your suggestion!

Comment (TobyRoseman) [reply src/python/turicreate/toolkits/style_transfer/_sframe_loader.py]: @dhivyaaxim - The new comment is not accurate. Python 3.7 does support raising a `StopIteration`. It just converts it to a `RuntimeError`.

**Pull request merged**